aws-lc-sys/CMakeLists.txt

```diff
@@ -10,43 +10,58 @@ enable_language(C)
 
 add_subdirectory(aws-lc aws-lc EXCLUDE_FROM_ALL)
 
-add_definitions(-DAWS_LC_RUST_INCLUDE_SSL)
+if (BUILD_LIBSSL)
+    add_definitions(-DAWS_LC_RUST_INCLUDE_SSL)
+endif()
 add_library(rust_wrapper STATIC rust_wrapper.c)
 target_include_directories(rust_wrapper PRIVATE include)
-target_link_libraries(rust_wrapper PUBLIC crypto ssl)
+
+if (BUILD_LIBSSL)
+    target_link_libraries(rust_wrapper PUBLIC crypto ssl)
+else()
+    target_link_libraries(rust_wrapper PUBLIC crypto)
+endif()
 
 set(FINAL_ARTIFACTS_DIRECTORY ${CMAKE_BINARY_DIR}/artifacts)
 
 # Places the compiled library files at the root of the binary directory so we can have a consistent
 # location to find the artifacts cross-platform.
-set_target_properties(rust_wrapper crypto ssl
+set_target_properties(rust_wrapper crypto
         PROPERTIES
         ARCHIVE_OUTPUT_DIRECTORY ${FINAL_ARTIFACTS_DIRECTORY}
         LIBRARY_OUTPUT_DIRECTORY ${FINAL_ARTIFACTS_DIRECTORY})
 
+if (BUILD_LIBSSL)
+    set_target_properties(ssl
+            PROPERTIES
+            ARCHIVE_OUTPUT_DIRECTORY ${FINAL_ARTIFACTS_DIRECTORY}
+            LIBRARY_OUTPUT_DIRECTORY ${FINAL_ARTIFACTS_DIRECTORY})
+endif()
+
 # Based on https://stackoverflow.com/a/7750816 as some generators, like MSVC, will try to prefix the output directory
 # which is not needed in our case.
 foreach (OUT_NAME ${CMAKE_CONFIGURATION_TYPES})
     string(TOUPPER ${OUT_NAME} OUT_NAME)
-    set_target_properties(rust_wrapper crypto ssl
+    set_target_properties(rust_wrapper crypto
             PROPERTIES
             ARCHIVE_OUTPUT_DIRECTORY_${OUT_NAME} ${FINAL_ARTIFACTS_DIRECTORY}
             LIBRARY_OUTPUT_DIRECTORY_${OUT_NAME} ${FINAL_ARTIFACTS_DIRECTORY})
+    if (BUILD_LIBSSL)
+        set_target_properties(ssl
+                PROPERTIES
+                ARCHIVE_OUTPUT_DIRECTORY_${OUT_NAME} ${FINAL_ARTIFACTS_DIRECTORY}
+                LIBRARY_OUTPUT_DIRECTORY_${OUT_NAME} ${FINAL_ARTIFACTS_DIRECTORY})
+    endif()
 endforeach ()
```

Comment:
Does this actually need to be modified? All this does is change the properties of a target, but it should only be compiled if the top level rust_wrapper is linked to it via `target_link_libraries`.

Reply:
The next commit simplifies the logic using a [function](https://cmake.org/cmake/help/latest/command/function.html). Thanks for the suggestion!